### PR TITLE
fix(core): property coerce for all packages

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
This adds a consistent utility function for converting a value to boolean. Pulling this back to v5 and to clr/angular to help ensure the boolean type coercion behavior is consistent between the libraries.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by hippee-lee — APPROVED** [review]

File: packages/core/src/internal/utils/dom.spec.ts [hunk 1]
```diff
@@ -833,4 +834,15 @@ describe('Functional Helper: ', () => {
...
   });
+
+  describe('coerceBooleanProperty():', () => {
+    it('should coerse a value to a boolean', () => {
+      expect(coerceBooleanProperty(null)).toBe(false);
+      expect(coerceBooleanProperty(undefined)).toBe(false);
+      expect(coerceBooleanProperty(false)).toBe(false);
+      expect(coerceBooleanProperty(true)).toBe(true);
+      expect(coerceBooleanProperty('false')).toBe(false);
+      expect(coerceBooleanProperty('true')).toBe(true);
```
Comment (mathisscott) [inline packages/core/src/internal/utils/dom.spec.ts:845]: probably need a test for empty string just to have it

File: packages/core/src/internal/utils/dom.ts [hunk 2]
```diff
@@ -238,3 +238,10 @@ export function getShadowRootOrElse(hostEl: HTMLElement, fallbackEl?: HTMLElemen
   const fallTo = fallbackEl ? fallbackEl : hostEl;
   return (hostEl.shadowRoot ? hostEl.shadowRoot : fallTo) as HTMLElement;
 }
+
+export type BooleanProperty = string | boolean | null | undefined;
+
+/** Coerces attribute/property value to a boolean */
+export function coerceBooleanProperty(value: any): boolean {
+  return value !== null && value !== undefined && `${value}` !== 'false';
```
Comment (mathisscott) [inline packages/core/src/internal/utils/dom.ts:246]: Making sure that empty string being true is expected. That's what will happen. I believe that is expected but probably want it in the unit tests to be explicit about it.

**Review by mathisscott — APPROVED** [review]

File: packages/core/src/internal/utils/dom.spec.ts (hunk 1 above)
Comment (coryrylan) [reply packages/core/src/internal/utils/dom.spec.ts:845]: ah saw this after the merge, github was stale I guess. I'll add one to be safe with a follow up PR
